public/app/core/services/keybindingSrv.ts

```diff
@@ -47,9 +47,36 @@ export class KeybindingSrv {
       this.bind('s o', this.openSearch);
       this.bind('f', this.openSearch);
       this.bind('esc', this.exit);
+      this.bindGlobal('esc', this.globalEsc);
     }
   }
 
+  globalEsc() {
+    const anyDoc = document as any;
+    const activeElement = anyDoc.activeElement;
+
+    // typehead needs to handle it
+    const typeaheads = $('.slate-typeahead--open');
```

Comment:
I have mixed feelings/don't have strong opinion about this PR. I know the esc to blur the input field was always there, but I don't think it's expected(browser default) behaviour. Especially that we now introduce this logic here which is quite error prone if only somebody changes typeahead's class name.

Also I think the behaviour of the esc should belong to the component interested in it rather that global scope.

Reply:
this logic was there before I did not add it. 

I assume ESC works in all model / edit workflows and find it super annoying that it does not anymore. It works in most software & modals so I find removing this was a bad idea.

Reply:
Ah, wait, so when we removed the global bind, we also removed the ESC handling for modals etc? I think we only removed that from the Edit mode

Reply:
Well it disabled it for all modals that have an input field which happens to have focus.